states/dedup/targetinput.go

```diff
@@ -120,9 +120,31 @@ type BuildArgInput struct {
 	DefaultValue string `json:"defaultConstant"`
 }
 
+// BuiltinVariables is a set of all the builtin variable names.
+var BuiltinVariables = map[string]bool{
+	"EARTHLY_TARGET":                  true,
+	"EARTHLY_TARGET_PROJECT":          true,
+	"EARTHLY_TARGET_NAME":             true,
+	"EARTHLY_TARGET_TAG":              true,
+	"EARTHLY_TARGET_TAG_DOCKER":       true,
+	"EARTHLY_GIT_HASH":                true,
+	"EARTHLY_GIT_BRANCH":              true,
+	"EARTHLY_GIT_TAG":                 true,
+	"EARTHLY_GIT_ORIGIN_URL":          true,
+	"EARTHLY_GIT_ORIGIN_URL_SCRUBBED": true,
+	"EARTHLY_GIT_PROJECT_NAME":        true,
+	"TARGETPLATFORM":                  true,
+	"TARGETOS":                        true,
+	"TARGETARCH":                      true,
+	"TARGETVARIANT":                   true,
+}
```

Comment:
Not sure what your suggestion is? Is it purely style-related? (we've generally typed sets as `map[..]bool` across our codebase so far, mainly for ergonomics purposes - but don't mind switching to `map[..]struct{}` if we wanted to)

> Removes the hard requirement for updaing in both places

This is the part I don't really understand. Not sure what you mean here?

Reply:
More or less just saying treat it like a set; where the presence of the builtin implies `true`, so then we dont have to remember to update both places? (or am I misunderstanding something about why we need to update the builtin list in both places?)

Reply:
We anyway assume there can only be true values in it. I don't think it would achieve what you have in mind.

We have two places where we have builtin build args because one is used to populate the actual value, and the other is used in the detection of whether an arg has default value or not. (Builtin args are deemed as always having a default value as they cannot be overridden). Default value detection is used for deduplicating target conversion.

I've put in a for loop to check that any builtin value that we set is also in the hard-coded set, so we don't forget about updating it.

Reply:
I've merged this to make progress - but happy to change after-the-fact in case I'm misunderstanding what you have in mind.